server/src/main/java/org/opensearch/action/admin/cluster/allocation/ClusterAllocationExplainAction.java

```diff
@@ -47,4 +50,9 @@ public class ClusterAllocationExplainAction extends ActionType<ClusterAllocation
     private ClusterAllocationExplainAction() {
         super(NAME, ClusterAllocationExplainResponse::new);
     }
+
+    @Override
+    public List<Scope> allowedScopes() {
+        return List.of(ActionScopes.Cluster_Read, ActionScopes.Cluster_ALL, ActionScopes.ALL);
```

Comment:
I left an explanation of the choice not to use a hierarchical permission model below. Let me know if that does not clarify. The idea is that one would need one of allowedScopes for the action to be allowed.

As to the lack of Cluster_ReadWrite, I went through all the actions and made my best judgement on what 'type' of action they were. I did not see any obvious actions under `**/action/admin/cluster` or the `**/action/*` level that suggested a need for writing at a cluster level. It is not particularly clear to me what that would look like.

The Cluster_ALL scope has all of the Cluster-level actions. So things beyond just reading cluster level information.

Reply:
What kind of application should have that access to `Cluster.All` instead of the full trust scope e.g. `Application.SuperUserAccess.All`?  I'd rather fewer more exclusive scopes if possible.

Extension point map cleanly into this there space, either its an ActionPlugin or it isn't, there is no way to define/register actions otherwise.